packages/chip-input/src/ux-choice-attribute.ts

```diff
@@ -0,0 +1,45 @@
+import { UxChoiceContainerAttribute } from './ux-choice-container-attribute';
+import { customAttribute, inject, observable, Optional } from 'aurelia-framework';
+import './ux-choice.css';
+
+@customAttribute('ux-choice')
+@inject(Element, Optional.of(UxChoiceContainerAttribute))
+export class UxChoiceAttribute {
+
+  public value: any;
+  @observable public selected: boolean;
+
+  constructor(private element: Element, private container: UxChoiceContainerAttribute) {}
+
+  public bind() {
+    this.container.registerChoice(this);
+  }
+
+  public attached() {
+    this.element.classList.add('ux-choice');
+  }
+
+  public detached() {
+    this.container.disposeChoice(this);
+  }
+
+  public valueChanged(newValue: any, oldValue: any) {
+    if (!this.container) {
+      return;
+    }
+    if (this.container.isMultiple && Array.isArray(this.container.value)) {
+      for (const index in this.container.value) {
```

Comment:
if it's an array then let's loop normally via index. Looping an array by `for ... in` could be not ideal

Reply:
also here we are repeating the property access chain quite a lot (.container, .value), maybe assign this to a local variable to save some cost?